src/gui/settings.py

```diff
@@ -60,12 +61,25 @@ def surveyEdit(self):
             vl = self.session.transectLayer
         elif self.sender() == self.strateButton:
             vl = self.session.strateLayer
-        elif self.sender() == self.plateformButton:
-            vl = self.session.plateformLayer
+        elif self.sender() == self.boatButton:
+            vl = self.session.boatLayer
         vl.startEditing()
         if not vl.featureCount():
             feat = QgsVectorLayerUtils.createFeature(vl)
             vl.addFeature(feat)
+        if vl == self.session.boatLayer:
+            self.session.plateformLayer
+            dlg = QDialog(self)
+            dlg.setWindowTitle(vl.name())
+            table = iface.showAttributeTable(vl)
+            originDlg = table.parent()
+            table.setParent(dlg)
+            if originDlg:  # version < 3.28 compatibility
+                originDlg.hide()
+            dlg.show()
+            dlg.destroyed.connect(vl.commitChanges)
+            dlg.destroyed.connect(self.session.plateformLayer.commitChanges)
+            return
```

Comment:
the early return is wanted?

Reply:
Yes, for the other layer, we open the first (and only) feature form. For the boat layer, we open the attribute table.